chain/client/src/client.rs

```diff
@@ -1673,14 +1673,15 @@ impl Client {
                     shard_id,
                     is_forwarded
                 );
-                self.shards_mgr.insert_transaction(shard_id, tx.clone());
 
                 // Active validator:
                 //   possibly forward to next epoch validators
                 // Not active validator:
                 //   forward to current epoch validators,
                 //   possibly forward to next epoch validators
                 if active_validator {
+                    self.shards_mgr.insert_transaction(shard_id, tx.clone());
+
                     if !is_forwarded {
                         self.possibly_forward_tx_to_next_epoch(tx)?;
                     }
```

Comment:
I'd also add a metrics for all the 3 cases here -- especially for 'NoResponse' one - as this one should be always equal to 0 (unless we have some bug in transaction routing)